AdvancementAPI.java

```diff
@@ -44,25 +56,37 @@ public String getIcon() {
         return icon;
     }
 
-    public AdvancementAPI withIcon(String icon) {
+    public MessengerAdvancement withIcon(String icon) {
         this.icon = icon;
         return this;
     }
 
+    public void setIcon(String icon) {
```

Comment:
About setters explained above - just to make the Api fit in cases when it's not necessary to have return
Though it's a thing to discuss